ccmlib/node.py

```diff
@@ -671,20 +671,19 @@ def stop(self, wait=True, wait_other_notice=False, other_nodes=None, gently=True
                 for node, mark in marks:
                     node.watch_log_for_death(self, from_mark=mark)
             else:
-                time.sleep(.1)
+                time.sleep(0.1)
 
             still_running = self.is_running()
             if still_running and wait:
-                # The sum of 7 sleeps starting at 1 and doubling each time
-                # is 2**7-1 (=127). So to sleep an arbitrary wait_seconds
-                # we need the first sleep to be wait_seconds/(2**7-1).
-                wait_time_sec = wait_seconds/(2**7-1.0)
-                for i in xrange(0, 7):
+                wait_time_sec = 0.1
+                elapsed = 0.0
+                while elapsed <= wait_seconds:
                     # we'll double the wait time each try and cassandra should
                     # not take more than 1 minute to shutdown
                     time.sleep(wait_time_sec)
                     if not self.is_running():
                         return True
+                    elapsed += wait_time_sec
```

Comment:
instead of calculating it, we can compare to a `start_time = time.time()`
With this in mind, the last sleep period can be trimmed to `start_time + wait_seconds`, otherwise, since it's double in every iteration, it can overshoot quite a bit.

Reply:
See `wait_until_stopped`